app.py

```
@@ -335,6 +350,22 @@ def upload_files_proc(upload_files):
     geometry_aoi_list = []
 
     for upload_file in upload_files:
+        file_name = getattr(upload_file, 'name').lower()
+        upload_file.seek(offset=0)
+
+        # Parse CSV file
+        if file_name.endswith(suffix='.csv'):
+            try:
+                csv_geom = parse_csv_coordinates(file=upload_file)
+                geometry_aoi_list.append(csv_geom)
+
+                # Update the last uploaded centroid
+                last_uploaded_centroid = csv_geom.centroid(maxError=1).getInfo()['coordinates']
+            except Exception as e:
+                print(e)
+            continue
```

this catches all exceptions and   and try to print them but it actually swallows all errors which hides useful info from the user, so I can't get proper feedback from user side. there's already a dedicated issue for error handling. besides, one should use streamlit's error/exception callout messages for proper display.

---

don't catch exceptions?

---

Yes, don't for now. and mind the rest of the reviews. Good luck.